utils/pckgen

```diff
@@ -0,0 +1,559 @@
+#!/bin/bash
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+# To make additional packages creation possible
+# through this script all you need to do is:
+# - add 'case' option to set variable 'GENERATE_<NAME>=1'
+# - add function 'generate_<name>' to handle the task
+# - add 'DEPENDENCIES_<NAME>' array if needed
+# - update help!
+
+
+NAME="open-cas-linux"
+# TODO: change 'supported from' to 20.3
+SUPPORTED_FROM_VERSION="20.1"
+THIS="$(basename $0)"
+ARCH="$(uname -i)"
+SCRIPT_BASE_DIR="$(dirname $(realpath $0))"
+DEFAULT_OUTPUT_DIR="$SCRIPT_BASE_DIR/packages"
+DEFAULT_BASE_SPEC="$SCRIPT_BASE_DIR/$NAME.spec.base"
+RPM_BUILD_DIR=~/rpmbuild
+RPM_SOURCES_DIR="$RPM_BUILD_DIR/SOURCES"
+RPM_SPECS_DIR="$RPM_BUILD_DIR/SPECS"
+RPM_RPMS_DIR="$RPM_BUILD_DIR/RPMS"
+RPM_SRPMS_DIR="$RPM_BUILD_DIR/SRPMS"
+TEMP_DIR="/tmp/$THIS"
+DEPENDENCIES=(git)
+# Dependencies for particular packages creation:
+DEPENDENCIES_TAR=(tar)
+DEPENDENCIES_ZIP=(zip)
+DEPENDENCIES_RPM=(tar wget rpmdev-setuptree rpmbuild)
+DEPENDENCIES_SRPM=("${DEPENDENCIES_RPM[@]}")
+# List of relative submodule directories:
+SUBMODULES=(
+    "ocf"
+    "test/functional/test-framework"
+)
+
+
+usage() {
+    echo "Usage:"
+    echo "  ./$THIS -t -z -r -s [options] <SOURCES_PATH>"
+    echo "  ./$THIS -v [options] <SOURCES_PATH>"
+    echo "  ./$THIS -C|-h"
+    echo "where SOURCES_PATH is an absolute path to the root directory of OpenCAS sources"
+}
+
+print_help() {
+    echo "Generate OpenCAS packages."
+    echo "$(usage)"
+    echo
+    echo "This script generates various OpenCAS packages like release archives (tar, zip),"
+    echo "RPMs (source and binary) as well as creates version file with metadata."
+    echo
+    echo "Mandatory arguments to long options are mandatory for short options too."
+    echo
+    echo "File creation:"
+    echo "  -t, --make-tar                  generate tar archive"
+    echo "  -z, --make-zip                  generate zip archive"
+    echo "  -r, --make-rpm                  generate RPM packages"
+    echo "  -s, --make-srpm                 generate SRPM package"
+    echo "  -v, --version-only              create only version file and exit;"
+    echo "                                  by default in 'SOURCES_PATH/metadata' directory,"
+    echo "                                  use -o option to redirect"
+    echo "  -M, --master                    if building packages, make them 'master' version;"
+    echo "                                  $THIS creates 'devel' version by default"
+    echo "  -R, --release                   if building packages, make them 'release' version;"
```

Comment:
We don't need this flag at all. We can easily detect it by checking if HEAD is tagged with release version. Same about master - if HEAD is merge request generated by GitHub, and there are no local changes, we can be pretty sure it's master package.